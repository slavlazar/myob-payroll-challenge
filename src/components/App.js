import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import { GlobalStyles } from './shared/ui';
import Login from '../containers/Login';
import Logout from '../containers/Logout';
import Payslip from '../containers/Payslip';

import NotFound from './NotFound';
import PrivateRoute from './shared/PrivateRoute';

import { LOGIN_URL } from '../utils/constants';

const App = () => (
  <React.Fragment>
    <GlobalStyles />
    <Router>
      <Switch>
        <Route path={LOGIN_URL} component={Login} />
        <PrivateRoute path="/logout" component={Logout} />
        <PrivateRoute path="/" exact component={Payslip} />
        <Route component={NotFound} />
      </Switch>
    </Router>
  </React.Fragment>
);

export default App;
