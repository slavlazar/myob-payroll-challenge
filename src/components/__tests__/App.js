import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

describe('Component: App', () => {
  it('renders correctly', () => {
    const wrapped = shallow(<App />);

    expect(wrapped).toMatchSnapshot();
  });
});
