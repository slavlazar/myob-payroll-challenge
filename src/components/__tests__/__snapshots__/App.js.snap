// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App renders correctly 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        path="/login"
      />
      <Connect(PrivateRouteComponent)
        component={[Function]}
        path="/logout"
      />
      <Connect(PrivateRouteComponent)
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</Fragment>
`;
